pynicotine/gtkgui/search.py

```diff
@@ -1596,6 +1606,26 @@ def on_filter_entry_changed(self, widget):
         if not widget.get_text():
             self.on_refilter()
 
+    def on_filter_entry_icon_press(self, entry, _icon_pos, press_event, *_args):
+
+        if press_event.triggers_context_menu():
```

Comment:
This is not compatible with GTK 4. https://docs.gtk.org/gtk4/signal.Entry.icon-press.html

Reply:
The method is present in both GTK 3 and 4:

https://docs.gtk.org/gdk3/method.Event.triggers_context_menu.html
https://docs.gtk.org/gdk4/method.Event.triggers_context_menu.html

Reply:
Yes, but the event is no longer exposed in GTK 4.

Reply:
I see. The intention was to allow provision for a secondary action, but it isn't really needed anyway.